jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceFactory.java

```
@@ -499,25 +505,40 @@ default List<Resource> split(String str, String delims)
                 }
                 else
                 {
-                    // Simple reference
-                    list.add(newResource(reference));
+                    // Simple reference. Could have a scheme like jar:file:, or just file:.
+                    // Or it could be a relative reference, in which case we need to
+                    // ensure it is absolute. Otherwise, comparisons between Resources
+                    // that point to the same resource but one is relative and one is absolute
+                    // will fail.
+                    if (URIUtil.hasScheme(reference))
+                    {
+                        // Could be a jar:file url, ensure it is unwrapped back to the file
+                        // otherwise it will be a MountedPathResource and consume a mount point
+                        // that might be unnecessary - the caller should always decide whether to mount
+                        URI uri = new URI(reference);
+                        if (unwrap)
+                            list.add(newResource(URIUtil.unwrapContainer(uri)));
+                        else
+                            list.add(newResource(uri.toASCIIString()));
+                    }
+                    else
+                    {
+                        Path p = Paths.get(reference);
+                        if (!p.isAbsolute() && LOG.isDebugEnabled())
+                            LOG.warn("Non-absolute path: {}", reference);
```

throw rather than warn?

---

I think it's caller dependent - `MetaInfConfiguration` will have a problem if it's not an absolute path, but other callers may be ok with that.

---

It should always be absolute IMO.
How about ...

``` java
   Path p = Path.of(reference).toAbsolutePath(); // enforce absolute path.
```

---

If you look at the line below the `throw` you will see that we always convert to absolute path, so I think your concern is already addressed.